### DISCUSSION TITLE
Schema is not displayed when implementing Depends with Request in FastAPI with manual parsing payload

### DISCUSSION THREAD

**Discussion by albertalexandrov** [fastapi/fastapi#10069] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends
from pydantic import BaseModel, EmailStr
from starlette.requests import Request

app = FastAPI()


class UserCreateData(BaseModel):
    first_name: str
    last_name: str
    email: EmailStr  # unique
    username: str  # unique


async def validate_user_create_data(request: Request) -> UserCreateData:
    body = await request.body()
    user_create_data = UserCreateData.parse_raw(body)

    # complex validation with nested models

    return user_create_data


@app.post('/users')
async def create_user(user_create_data: UserCreateData = Depends(validate_user_create_data)):
    ...
```

### Description

Hi!

With code above schema `UserCreateData` is not displayed in swagger.

![image](https://github.com/tiangolo/fastapi/assets/23693954/1deb8a2a-2070-406b-8eb0-5e560e22f354)

Could you please tell me how to fix it?

 

### Operating System

macOS

### FastAPI Version

0.98

### Pydantic Version

1.10

### Python Version

3.10

**Comment by jgould22** [comment, marked as answer]
`validate_user_create_data` takes Request which can be anything so fastapi doesnt know what to put there.

You can probably get what you want by[ specifying it explicitly in examples.](https://fastapi.tiangolo.com/tutorial/schema-extra-example/?h=openap#openapis-examples-field)

You may also be able to use[ pydantic's validation ](https://docs.pydantic.dev/latest/usage/validators/)to validate your type and then just accept `UserCreateData` on your endpoint [as they do in the examples.](https://fastapi.tiangolo.com/tutorial/body/#import-pydantics-basemodel)
